query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/commons/LdapServiceCommons.java

```diff
@@ -241,14 +241,15 @@ private LdapConfigurationField ldapLoginServiceToLdapConfiguration(Map<String, O
         .pid(mapValue(props, SERVICE_PID_KEY));
   }
 
+  @SuppressWarnings("squid:S1135" /* Remove when TODO is completed */)
   private LdapConnectionField getLdapConnectionField(
       Map<String, Object> props, String ldapUrl, String startTls) {
     LdapConnectionField connection = new LdapConnectionField();
     URI ldapUri = getUriFromProperty(mapValue(props, ldapUrl));
 
     if (ldapUri != null && ldapUri.getScheme() != null) {
       // TODO: tbatie - 8/17/17 - It'd be great if we had some sort of match method in the EnumValue
-      // instead of doing little checks like this
+      // instead of doing little checks like this. Remove sonar suppression when task is completed.
```

Comment:
Can we confirm this TODO is still relevant? There was a match check done added to the BaseEnum field.

Reply:
`tbatie` added that TODO

Reply:
You could override the `setValue` method and do some smart logic there. I'm fine with skipping this, it feels a bit beyond the scope of this PR. https://github.com/connexta/admin-console/blob/725f1d860ebdab0e42b741cd943f63a638c2d3d5/query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/fields/connection/LdapEncryptionMethodField.java